packages/flutter/lib/src/material/tabs.dart

```diff
@@ -476,10 +476,21 @@ class _TabBarScrollPosition extends ScrollPositionWithSingleContext {
 
   final _TabBarState tabBar;
 
+  bool _hadNonZeroDimension = false;
+
   @override
   bool applyContentDimensions(double minScrollExtent, double maxScrollExtent) {
     bool result = true;
-    if (pixels == null) {
+    if (!_hadNonZeroDimension) {
```

Comment:
This is good. I found it slightly difficult to read because the initial value of _hadNonZeroDimension seemed to guess the future and `!_hadNonZeroDimension` required some unpacking. 

Here's another way to encode this:
```
  bool _initialViewportDimensionWasZero;

  if (_initialViewportDimensionWasZero != true) {
      // If the viewport never had a non-zero dimension, we just want to jump
      // ...
      assert(viewportDimension != null);
      _initialViewportDimensionWasZero = viewportDimension == 0.0;
      ...
  }
```

If you don't find this clearer, I have no problem with leaving the code as it is. Just a thought.

Reply:
Changed it as per your suggestion.